### PR TITLE
Check to see if _s theme has actually been created

### LINKED ISSUE #1698: scaffold _s not creating the theme but saying it did
I run ` wp scaffold _s test_theme --theme_name="Test Theme" --author="Test Author" --author_uri="https://example.es/"`, I get `Success: Created theme 'Test Theme'.`, but no theme files are created anywhere I see.
My file structure is this:
```
- root
--- wordpress
--- content
----- themes
----- plugins
```
I have an `index.php` file in root, which only defines WP_USE_THEMES as `true`, and with `require( 'wordpress/wp-blog-header.php' );`

Then my wp-config.php is in root as well, where I have (besides the other necessary bits):
```
define( 'WP_HOME', 'http://local.domain' );
define( 'WP_SITEURL', WP_HOME . '/wordpress' );
define( 'WP_CONTENT_DIR', realpath( $_SERVER['DOCUMENT_ROOT'] . '/content' ) );
define( 'WP_CONTENT_URL', WP_HOME . '/content' );
```

This is the structure I use for all my sites, and it works. wp-cli usually works as well with this. Just wondering if the scaffolding command wasn't supported in a structure like this? Also, why would it say the theme was created if nothing is actually created? Could it be it is creating the files somewhere else? Couldn't find them anywhere, and I'm not knowlegeable enough to debug the phar file to see on which step something is going amiss.

It's a single site clean install, no plugins. `local.domain` resolves to localhost, obviously.

Thanks in advance for any help, and regards.

WP 4.1.1, OSX Yosemite.

### PR DESCRIPTION
**Pull request by yivi**
Checks the result of the unzip operartion, and displays Error message if failed; attempting to give some troubleshooting info. Resolves #1698

### CODE REVIEW

File: php/commands/scaffold.php [hunk 1]
```diff
@@ -114,9 +114,9 @@ private function _scaffold( $slug, $assoc_args, $defaults, $subdir, $templates )
 
 		$control_args = $this->extract_args( $assoc_args, array(
 			'label'  => preg_replace( '/_|-/', ' ', strtolower( $slug ) ),
-			'theme'  => false,
-			'plugin' => false,
-			'raw'    => false,
+			'theme'  => FALSE,
+			'plugin' => FALSE,
+			'raw'    => FALSE,
```
Comment (danielbachhuber) [inline php/commands/scaffold.php]: Can you change these back to lowercase?

File: php/commands/scaffold.php [hunk 2]
```diff
@@ -227,13 +232,19 @@ function _s( $args, $assoc_args ) {
 		$this->maybe_create_themes_dir();
 
 		$this->init_wp_filesystem();
-		unzip_file( $tmpfname, $theme_path );
+		$unzip_result = unzip_file( $tmpfname, $theme_path );
 		unlink( $tmpfname );
 
-		WP_CLI::success( "Created theme '{$data['theme_name']}'." );
+		if ( TRUE === $unzip_result ) {
```
Comment (danielbachhuber) [inline php/commands/scaffold.php]: Lowercase please

File: php/commands/scaffold.php [hunk 3]
```diff
@@ -227,13 +232,19 @@ function _s( $args, $assoc_args ) {
...
+			WP_CLI::success( "Created theme '{$data['theme_name']}'." );
+		} else {
+			WP_CLI::error( "Could not decompress your theme files ('$tmpfname') at '$theme_path': " . $unzip_result->get_error_message() );
```
Comment (danielbachhuber) [inline php/commands/scaffold.php]: If you're including variables in a string, can you wrap them in braces? `'{$tmpfname}'`

File: php/commands/scaffold.php [hunk 4]
```diff
@@ -590,7 +606,8 @@ private function pluralize( $word ) {
 				return preg_replace( $rule, $replacement, $word );
 			}
 		}
-		return false;
+
+		return FALSE;
```
Comment (danielbachhuber) [inline php/commands/scaffold.php]: Same as previous

**Comment by danielbachhuber** [comment]
Thanks! Can you add [functional tests](https://github.com/wp-cli/wp-cli/blob/master/CONTRIBUTING.md#functional-tests) to this pull request, reproducing your original reporting steps?

**Comment by yivi** [comment]
Sorry about all the uppercase booleans. My IDE is set to autoformat on save and I didn't realize I had changed things all around the file... :P

I don't know anything about functional tests. I'll try to read some later tonight to see if I can whip up something for tomorrow. Should I include my tests in `scaffold.feature`? (If I manage to understand things well enough)

**Comment by danielbachhuber** [comment]
> Should I include my tests in scaffold.feature?

Yep, that's a good place for them.

**Comment by danielbachhuber** [comment]
@yivi Will you be able to complete this?

**Comment by yivi** [comment]
Sorry Daniel, had no time yet. Tomorrow Saturday I'll spend some time
reading about tests and trying to finish this, or I'll tell you if I find I
can't do it in time.

Regards,

2015-04-02 14:52 GMT+02:00 Daniel Bachhuber <notifications@github.com>:

**Comment by yivi** [comment]
@danielbachhuber , finished with the functional test for this check.
Hopefully it's what you were looking for, and the test is not too simplistic.

Since I think that there could be other scenarios where the unzip could fail (permissions come to mind), rather than trying to think all the possible situations that would bring a failed outcome I'm just setting wp_content_dir so the following unzip operation will generally fail, and hopefully cover more ground that way.

Regards

**Comment by danielbachhuber** [comment]
@yivi Can you merge master?

**Comment by yivi** [comment]
Merged, @danielbachhuber 

There are a couple tests that fail on my machine:
import.feature:43
post.feature:32
user.feature:203

But they don't aren't related to this PR, and they seem to fail for small output differences...

Regards,

**Pull request merged**